client/src/main/scala/org/apache/celeborn/client/commit/ReducePartitionCommitHandler.scala

```diff
@@ -86,20 +82,26 @@ class ReducePartitionCommitHandler(
     inProcessStageEndShuffleSet.remove(shuffleId)
   }
 
-  override def finalCommit(
-      shuffleId: Int,
-      recordWorkerFailure: ShuffleFailedWorkers => Unit): Unit = {
-    if (stageEndShuffleSet.contains(shuffleId)) {
+  override def canDoFinalCommit(shuffleId: Int): Boolean = {
```

Comment:
We can refactor the finalCommit to tryFinalCommit which only returns true if it really triggers commit, instead of using another canDoFinalCommit method.

Reply:
Good idea!